sway-core/src/ir_generation/function.rs

```diff
@@ -485,7 +485,12 @@ impl FnCompiler {
             Intrinsic::AddrOf => {
                 let exp = arguments[0].clone();
                 let value = self.compile_expression(context, md_mgr, exp)?;
-                Ok(self.current_block.ins(context).addr_of(value))
+                let span_md_idx = md_mgr.span_to_md(context, &span);
```

Comment:
Thanks.  Looks like this could be called once at the top and shared between variants.  But that can happen next time.